adagrams/game.py

```diff
@@ -58,74 +61,65 @@
+        letters.append(available_pool.pop(random_index))  #update the pool
+    
+    return letters
 
- 
 
 
 def uses_available_letters(word, letter_bank):
     status = False
-    #Can we use .copy()?
-    #Can we use lower() or capitalize()?
     count_of_letter_mapping = {}
 
-    for i in letter_bank:
-        if i in count_of_letter_mapping:
-            count_of_letter_mapping[i]+= 1
+    for letter in letter_bank:
+        if letter in count_of_letter_mapping:
+            count_of_letter_mapping[letter]+= 1
         else:
-            count_of_letter_mapping[i]= 1
-
-    #print(count_of_letter_mapping)
+            count_of_letter_mapping[letter]= 1
 
-    for i in word:
-        if i.capitalize() in letter_bank and count_of_letter_mapping[i.capitalize()]>0:
-            count_of_letter_mapping[i.capitalize()] -= 1
+    for letter in word:
+        if letter.upper() in letter_bank and count_of_letter_mapping[letter.upper()]>0:
+            count_of_letter_mapping[letter.upper()] -= 1
             status = True
         else:
             status = False
+            return status  #add a return statement so that we return False immediately when seeing invalid letter
     return status
 
 
 def score_word(word):
     score = 0
-    for i in word:
-        score += SCORE_OF_LETTER[i.capitalize()]
-    if len(word) >= 7:
-        score += 8
+    for letter in word:
+        score += SCORE_OF_LETTER[letter.upper()]
+    if len(word) >= BONUS_MIN_LENGTH:
+        score += LENGTH_BONUS_POINTS
     return score
 
 def get_highest_word_score(word_list):
     highest_score = 0
     winner = ""
-    for i in word_list:
-        if score_word(i) > highest_score:
-            highest_score = score_word(i)
-            winner = i
-        elif score_word(i) == highest_score:
-            if len(i) < len(winner):
-                winner = i
-
-    for i in word_list:
-        if len(i) == 10:
-            winner = i
-            break
+    for word in word_list:
+        score_of_word = score_word(word)
+        if score_of_word > highest_score:
+            highest_score = score_of_word
+            winner = word
+        #Updated tie condition
+        elif score_of_word == highest_score:
+            if len(winner) != 10 and len(word) == 10:  #First check if the given word has 10 letters since it's preferable under tie condition
+                winner = word                              #Also check if the winner word already has 10 letters so that when multiple words having same length same score, we pick the first word.
+            elif len(winner) != 10 and len(word) < len(winner):  #If the given word doesn't have 10 letters, check if this word has fewer letters. 
+                winner = word                                        #Also need to check if the winner has 10 letters because we don't want to acdientally update the winner to word that has fewer letter (again, 10 letters is the most perferable under tie condition)
```

Comment:
Notice that both of these conditions use the same partial condition in their expressions (checking that the best word so far is not already of length 10). Rather than listing it on each nested condition, we could move this up to the parent condition, simplifying each nested condition. This might make them understandable enough that we could combine them into a single condition, since they each have the same block logic.